iroh-relay/src/client/conn.rs

```diff
@@ -17,22 +16,50 @@ use tokio_tungstenite_wasm::WebSocketStream;
 use tokio_util::codec::Framed;
 use tracing::debug;
 
-use super::KeyCache;
-use crate::protos::relay::{ClientInfo, Frame, MAX_PACKET_SIZE, PROTOCOL_VERSION};
+use super::{ConnectError, KeyCache};
+use crate::protos::{
+    self,
+    relay::{ClientInfo, Frame, MAX_PACKET_SIZE, PROTOCOL_VERSION},
+};
 #[cfg(not(wasm_browser))]
 use crate::{client::streams::MaybeTlsStreamChained, protos::relay::RelayCodec};
 
 /// Error for sending messages to the relay server.
 #[derive(Debug, thiserror::Error)]
 pub enum ConnSendError {
     /// An IO error.
-    #[error("IO error")]
+    #[error(transparent)]
     Io(#[from] io::Error),
     /// A protocol error.
     #[error("Protocol error")]
     Protocol(&'static str),
 }
 
+/// Error when handshaking with the server
+#[derive(Debug, thiserror::Error)]
+pub enum HandshakeError {
+    /// An IO error.
+    #[error("failed to send handshake message to the relay")]
+    Send(#[from] protos::relay::Error),
+}
+
+/// Errors when receiving messages from the relay server.
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum RecvError {
+    /// An IO error.
+    #[error(transparent)]
+    Io(#[from] io::Error),
+    #[error(transparent)]
+    Protocol(#[from] crate::protos::relay::Error),
+    #[error(transparent)]
+    Websocket(#[from] tokio_tungstenite_wasm::Error),
+    #[error("invalid protocol message encoding")]
+    InvalidProtocolMessageEncoding,
+    #[error("Unexpected frame received {0}")]
+    UnexpectedFrame(crate::protos::relay::FrameType),
```

Comment:
`Websocket` seems like part of the `Io` variant?
And `InvalidProtocolMessageEncoding` and `UnexpectedFrame` are part of the `Protocol` variant?

I think the `io/protocol` distinction is pretty decent on a high-level and that was what I was going for in the `ConnSendError`.

Also it might be possible to have `ConnSendError` and `RecvError` be the same type?